src/utils/types.zig

```diff
@@ -324,14 +386,36 @@ fn sliceEql(comptime T: type, a: []const T, b: []const T) bool {
         .stage2_riscv64 => false,
         else => true,
     };
-    if (!@inComptime() and std.meta.hasUniqueRepresentation(T) and backend_can_use_eql_bytes)
+    if (!@inComptime() and
+        std.meta.hasUniqueRepresentation(T) and
+        backend_can_use_eql_bytes and
```

Comment:
These checks happen in `std.mem.eql`, not need to do it here.

Reply:
I intentionally copied these checks directly from `std.mem.eql`.

Let's say we call `std.mem.eql` here without doing these checks first, and for some type, one of these checks are false. std.mem.eql will use `==` to compare each element. What I actually wanted is to use my `eql` function in that case. These filters ensure the desired behavior. Basically all I wanted was a way to call `eqlBytes` in the same situation when `std.meta.eql` would, just with an added filter to avoid comparing pointers. Since `eqlBytes` is private, the easiest way to do that is by indirectly calling it through `std.meta.eql`.

I added a comment to explain this. Also I improved the pointer filter to actually check if there are pointers contained, rather than blindly skipping the optimization.

c2a717961e08d700c05435cb35677725c2e6eb27

Reply:
Ah, you're right, thanks for explaining it to me. I missed that bit of control flow. :+1: